.travis.yml

```diff
@@ -5,13 +5,6 @@ rust:
 - beta
 - stable
 - nightly
-matrix:
```

Comment:
@killercup @epage I removed the nightly-2017-03-15, are you ok with this?

Reply:
How come?

Reply:
The `.envs` methods from `Command` is stable since 1.10. But not for the `2017-03-15` build. I was wondering about why we are still testing this particular build.

Reply:
The reason (which I like and want to copy to my other projects) is to pin the version of rust/clippy.

In my other projects, we are always running the latest clippy.  Sometimes it is out of sync with rustc, so the build ignores clippy, losing coverage.  Even when it succeeds, there might be new clippy warnings which will get turned into errors, sporadically breaking our build.

What I'd recommend is to lock us to a new version of clippy.  This might be best as a separate PR so any clippy fixes don't get mixed in with your changes.

Reply:
@killercup I love having everything checked in the CI!

I'm unsure how stable rustfmt's output is from release to release.  I've not yet submitted a rustfmt.toml to my other projects.  I have played with a build optimization for rustfmt for locking to a specific version that you might find handy.

See https://github.com/cobalt-org/cobalt.rs/blob/master/.travis.yml#L29

Reply:
You're using the old rustfmt, not rustfmt-nightly :) I opened #48 for getting our formatting in check.